[blog] Announcing contribute.dev (#263)
gitpod-io/website: src/blog/frictionless-oss-contributions.md

---
subtitle: No Setup, Less Waiting, More Flow

> title: Frictionless Open-source Contributions

Author: Another thought, maybe `Frictionless Open-Source Contributions` can be the subtitle, and the title could be:

```
Announcing contribute.dev -- contribute to OSS in one click!
```

---
subtitle: Frictionless Open Source Contributions
title: Announcing contribute.dev — contribute to OSS in one click!
date: Mon Oct 7 2019 15:00:00 GMT+0000 (UTC)
author: nisarhassan12
image: https://user-images.githubusercontent.com/46004116/66325437-b11a9980-e940-11e9-9db4-69af48a0a52a.png
---

> <h3 style="margin: 1rem 0;">TL;DR</h3>

> > To celebrate Open Source and Hacktoberfest, we're stoked 🧙✨🥰 to announce <a href="https:// contribute.dev" target="blank">contribute.dev</a> It has a list of really cool Open Source projects and can all get you started coding with a single click!

Author: ```suggestion
> To celebrate Open Source and Hacktoberfest, we're stoked 🧙✨🥰 to announce <a href="https:// contribute.dev" target="blank">contribute.dev</a>. It has a list of really cool open-source projects that can get you started coding with a single click!
```

![Contribution Steps](https://user-images.githubusercontent.com/46004116/66287768-66713100-e8f0-11e9-9bb9-b712236059d0.png)

Before step 2 you have to make sure that you have [Node.js](https://nodejs.org/) installed and then you have to continue with the rest of the steps. Now the setup for Carbon is not that complex but it can be intimidating and frustrating for beginners and it still requires time and energy to be spent on what is not even part of the contribution.

Contrary to Carbon, other popular and valuable Open Source projects usually have more complicated development environments (i.e they include modern build pipelines + test suites + linters and things like that and so on). Configuring and setting up all this stuff can be cryptic, hard and can take a lot of time and effort.

What if something could save us from spending our time and energy on configuring all these cryptic and repetitive development setups? What if we could automate all of this with a single click, so that we can focus and spend more time on our goal which is contributing to Open Source, while forgetting about the development environment?

> Well! [contribute.dev](https://contribute.dev) is here to help we're stoked 🧙✨🥰 to announce it. It has a list of really cool Open Source Projects that are beginner-friendly and can all get you started with a single click:

Author: ```suggestion
Well! [contribute.dev](https://contribute.dev) is here to help, and we're stoked 🧙✨🥰 to announce it. It has a list of really cool Open Source Projects that are beginner-friendly and can all get you started with a single click:
```

---
subtitle: Frictionless Open Source Contributions

> title: Announcing contribute.dev — contribute to OSS in one click!

Author: No teaser image? 🙂

<img width="1016" alt="Screenshot 2019-10-07 at 16 19 12" src="https://user-images.githubusercontent.com/599268/66320046-638d3500-e91e-11e9-9ba6-915305faedd3.png">

Feel free to pick one -- I recommend https://unsplash.com/ as a source for great inspiring free images.

date: Mon Oct 7 2019 15:00:00 GMT+0000 (UTC)
author: nisarhassan12
---

> <div style="background: #123; padding: 20px 25px 1px; font-size: 90%;margin-bottom: 2rem;">

Author: Oof, that's quite blue:

<img width="764" alt="Screenshot 2019-10-07 at 16 22 01" src="https://user-images.githubusercontent.com/599268/66320173-9fc09580-e91e-11e9-9f1e-e7eded31775f.png">

And we don't do that anywhere else in our blog.

Usually, we use quotes instead:

<img width="769" alt="Screenshot 2019-10-07 at 16 23 29" src="https://user-images.githubusercontent.com/599268/66320276-d3032480-e91e-11e9-8b3a-01ca2f540f2b.png">

Please use a regular mardown quote for this text, by prefixing each line with `> ` instead of wrapping them in a custom-styled `<div>`.

    <h3 style="margin-top: 0;">TL;DR</h3>
    <p>To celebrate Open Source and Hacktoberfest, we're stoked to announce <a href="https://
    contribute.dev" target="blank">contribute.dev</a> It has a list of really cool Open Source projects and can all
    get you started coding with a single click!</p>
</div>

> Contributing to Open Source can be intimidating as you have to spend a lot of time while going through long README files to find out how to do the development setup that is essential for getting started. This is a serious barrier for contributors. It can be a huge pain especially if you're someone who is just starting out. Every day developers waste millions of hours and energy switching contexts between projects and branches, setting up development environments, or simply waiting for builds to complete when all they want is to start coding.

Author: I think you can make this text shorter and more impactful by deleting the last (complicated) phrase:

```suggestion
Contributing to Open Source can be intimidating as you have to spend a lot of time while going through long README files to find out how to do the development setup that is essential for getting started. This is a serious barrier for contributors. It can be a huge pain especially if you're someone who is just starting out.
```

Let's say you want to contribute to [Carbon](https://carbon.now.sh/) (a JS app that lets you create and share beautiful images of your source code). Before you can make your contribution you have to make sure that you have done everything from step 1 to 4 as shown below in the screenshot from the [Contribution Guide of Carbon](https://github.com/carbon-app/carbon/blob/master/.github/CONTRIBUTING.md).

![Contribution Steps](https://user-images.githubusercontent.com/46004116/66287768-66713100-e8f0-11e9-9bb9-b712236059d0.png)

> Before step 2 you have to make sure that you have [Node.js](https://nodejs.org/) installed and then you have to continue with the rest of the steps. Now the setup for Carbon is not that complex but it can be intimidating and frustrating for beginners and it still requires time and energy to be spent on what is not even the part of the contribution.

Author: Typo (unnecessary word "the"):

```suggestion
Before step 2 you have to make sure that you have [Node.js](https://nodejs.org/) installed and then you have to continue with the rest of the steps. Now the setup for Carbon is not that complex but it can be intimidating and frustrating for beginners and it still requires time and energy to be spent on what is not even part of the contribution.
```

Contrary to Carbon, other popular and valuable Open Source projects usually have more complicated development environments (i.e they include modern build pipelines + test suites + linters and things like that and so on). Configuring and setting up all this stuff can be cryptic, hard and can take a lot of time and effort.

> What if somehow something can save us from spending our time and energy from configuring and setting up all these cryptic and repetitive development setups? What if something can automate all of this with a single click so that we can focus and spend more time on our goal which is contributing to Open Source while forgetting about the development environment?

Author: Improvement suggestions:

```suggestion
What if something could save us from spending our time and energy on configuring all these cryptic and repetitive development setups? What if we could automate all of this with a single click, so that we can focus and spend more time on our goal which is contributing to Open Source, while forgetting about the development environment?
```

> Well! [contribute.dev](https://contribute.dev) is here to help we're stoked to announce it. It has a list of really cool Open Source Projects that are beginner-friendly and can all get you started with a single click!

Author: Maybe replace the exclamation point at the end? (We're excited, but maybe the tone can be a bit more neutral here. 😊)

```suggestion
Well! [contribute.dev](https://contribute.dev) is here to help we're stoked to announce it. It has a list of really cool Open Source Projects that are beginner-friendly and can all get you started with a single click:
```

![Projects List](https://user-images.githubusercontent.com/46004116/66307071-d695ac00-e91c-11e9-9eb1-8aa97bb6ca67.png)

> Setup for all of the listed projects is completely automated with the help of [Gitpod](https://gitpod.io) (an online IDE) within seconds, it provides you with a fully working development environment, including a VS Code-like interface and a cloud-based Linux container configured specifically for your project.

Author: Minor improvement:

```suggestion
Setup for all of the listed projects is completely automated with the help of [Gitpod](https://gitpod.io) (an online IDE) so within seconds, it provides you with a fully working development environment, including a VS Code-like interface and a cloud-based Linux container configured specifically for your project.
```

For getting started with the project of your choice just hit the blue `Open in Gitpod` button in the bottom right corner.

![image](https://user-images.githubusercontent.com/46004116/66307306-82d79280-e91d-11e9-9c8b-00d92b710892.png)

> In a matter of seconds this will launch a ready to code workspace with everything setup ( like the dependencies pre-installed and the web server running ) as shown below.

Author: Please remove spaces within brackets:

```suggestion
In a matter of seconds this will launch a ready to code workspace with everything setup (like the dependencies pre-installed and the web server running) as shown below.
```
